### PR TITLE
Add lemma and ent_type attributes to the merge_noun_chunks function

### PR DESCRIPTION
**Pull request by alaponin**
## Description

A PR for issue #4107 

### Types of change

I have included the following small enhancements:
The lemma of the merged token consists of the lemmas of all the tokens. 
The entity type of the merged token is the entity type of the root token.

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by svlandeg** [comment]
Hi @alaponin , thanks for the PR! Could you also write an appropriate unit test to check this behaviour? You could write it at `spacy/tests/regression/test_issue4107` and you could use `assert` statements to verify behaviour on a toy example that wasn't working before, but now does work after your fix.

For some inspiration, a vaguely related unit test checking whether vector information was properly copied after running `tokenizer.split` can be found [here](https://github.com/explosion/spaCy/blob/master/spacy/tests/regression/test_issue3540.py).

File: spacy/pipeline/functions.py [hunk 1]
```diff
@@ -16,7 +16,8 @@ def merge_noun_chunks(doc):
         return doc
     with doc.retokenize() as retokenizer:
         for np in doc.noun_chunks:
-            attrs = {"tag": np.root.tag, "dep": np.root.dep}
+            lemma = doc[np.start:np.end].lemma_
+            attrs = {"tag": np.root.tag, "dep": np.root.dep, "lemma": lemma, "ent_type": np.root.ent_type}
             retokenizer.merge(np, attrs=attrs)
```
Comment (svlandeg) [inline spacy/pipeline/functions.py]: Is this behaviour specific to the `merge_noun_chuncks` function, or should this be fixed in the underlying `retokenizer.merge` functionality?

Comment (honnibal) [reply spacy/pipeline/functions.py]: @svlandeg I do think we want to take care of this in the retokenizer. Noun chunks are always going to be a subtree, but for other merges that work as a subtree it's good to keep the parse working properly. I actually thought we already did this?

**Comment by ines** [comment]
This should now be resolved automatically by #4219, so I'm closing this! Thanks for the PR though – there was obviously nothing wrong with it, but it ended up inspiring a more general fix 🙂

**Pull request closed without merging**